### ISSUE TITLE
FreeAccount ergonomy

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Risehat/freemedforms#162]
```
This is my personal opinion... I think that there are many possibilities to 
improve the ergonomy of the account plugin.
Also, we have to improve its integration inside the FreeMedForms EMR.
```

Original issue reported on code.google.com by `eric.mae...@gmail.com` on 29 Aug 2012 at 11:02
- Blocked on: #194

**Comment by GoogleCodeExporter** [comment]
```
please specify the ergonomics procedures to improve.
I've just taken in consideration the ergonomic criticisms of C.Reiter and fixed 
them.
```

Original comment by `pm.desom...@gmail.com` on 29 Aug 2012 at 2:57
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
one thing I saw: the "Input" and "Control" radiobuttons are now at least 
reversible - you can go back to the previous state.
But the UI changes massively, and the radiobuttons are 1. not in the same 
place, 2. they are complete different objects (which is in code design very bad 
to debug, complicated, etc.)

What I would suggest is, instead of the radiobutton, which normally should be 
used when choosing 2 or more OPTIONS of a value, we should use stackedwidgets 
with tabs, because "Control" and "Input" are completely different modes to work 
with the accounting.
So a radiobutton is definitely the wrong widget for changing that mode.
```

Original comment by `christian.a.reiter@gmail.com` on 29 Aug 2012 at 3:49

**Comment by GoogleCodeExporter** [comment]
```
* receiptviewer.ui and controlreceipt could be one .ui file then. Don't know if 
there are otherdependencies, or if one of that ui is used separetedly.
* The whole "Actions" groubBox in Controlreceipts is bad placed. It could be 
completely removed, and the actions could be placed like in the patientbar, on 
the upper screen/widget dock - so the UI would be consistent across FMF.
```

Original comment by `christian.a.reiter@gmail.com` on 29 Aug 2012 at 3:54

**Comment by GoogleCodeExporter** [comment]
```
you don't like my radio buttons ;)
The pb of tab widgets is that it take too much space. 


```

Original comment by `pm.desom...@gmail.com` on 29 Aug 2012 at 4:01

**Comment by GoogleCodeExporter** [comment]
```
ok, but what patient bar ?
```

Original comment by `pm.desom...@gmail.com` on 29 Aug 2012 at 4:04

**Comment by GoogleCodeExporter** [comment]
```
Oh, sorry, I meant the user selector - there is a bar at the top that contains 
all the actions that can be taken. This would be clearer IMHO. See screenshot.
```

Original comment by `christian.a.reiter@gmail.com` on 29 Aug 2012 at 6:14

Attachments:
- [screenshot10.png](https://storage.googleapis.com/google-code-attachments/freemedforms/issue-162/comment-6/screenshot10.png)

**Comment by GoogleCodeExporter** [comment]
```
And about free space: the whole action groupbox would be gone then, in the 
"Filtering data" groupbox is plenty of space even now, and in the Input section 
there is plenty of space as well in the Dates groupbox and  "Choose your 
billing values".
So the toolbar would be really no problem regarding space...
```

Original comment by `christian.a.reiter@gmail.com` on 29 Aug 2012 at 8:02

**Comment by GoogleCodeExporter** [comment]
```
why not the way to.
But why is it more ergonomic ?
You don't like big buttons. However, big buttons are more ergonomic. It is 
easier to "get the target". 
With small buttons, you have to pay attention, and you cannot hear the patient 
in the same time.
With big buttons, you don't have the same problems.
There are plenty of softs with big buttons.
```

Original comment by `pm.desom...@gmail.com` on 30 Aug 2012 at 9:23

**Comment by GoogleCodeExporter** [comment]
```
oh - when coming to ergonomics, then buttons and mouse clicks are COMPLETELY 
bad.
There are three things to differ:
1. Consistent and appealing UI
2. Usability
3. Ergonomics or better *Efficiency*

They interfere but only overlap and are generally different.
So e.g. big buttons are easier to target, yes, but if you want to have 
efficiency, qou'd better don't use mouse grab, mouse move, click and buttons. 
Far faster and more efficient is using keyboard shortcuts. And then the button 
size is not important and can follow common and OS style principles.

```

Original comment by `christian.a.reiter@gmail.com` on 30 Aug 2012 at 11:31

**Comment by GoogleCodeExporter** [comment]
```
I didn't find the interview with an Ubuntu GUI designer. But this one was very 
important and interesting. He told about the difference of efficiency and 
usability, and why this both things often get mixed up in the heads of 
programmers.
An efficient program like all the old-school text-mode software that relied on 
the F1-F12 keys as shortcuts, If you still know theese - it was VERY fast and 
efficient, BUT you had to KNOW about every shortcut and command.
The new-style happy-shiny GUIs are very visually appealing, but extremely slow 
compared to that ones.
Usability, in the counterpart, is about designing the UI in a way so that any 
new user that never used this software before can use it without reading the 
100page manual before.


So how is the mix:
1. Usability: Make a good, consistent UI. Place the buttons (and everything 
else) where users expect them. (big buttons are no problem. A problem is 
sometimes big buttons, sometimes small buttons.) Make the interface intuitive. 
Apple does a very good job here. That's what I meant with the AccountPlugin: 
it's absolutely NOT easy to use from a usability's view.
2. Efficiency: Additionally make shortcuts (e.g. keyboard shortcuts or hidden 
tricks, menu items, etc.) that speed up the work ONCE YOU ARE ADAPTED TO THE 
SOFTWARE AND WANT TO GET FASTER.

So do you see what I mean?
I don't "don't like your buttons" ;-)
It's just that nowhere else in FMF are bug buttons in the UI, and it's not 
consistent - AND it has no advantage. Because "it's easier to target" already 
means you had to take the mouse with your hand, move it to the button and click 
it, instead of just maybe hitting "Enter", for. e.g the default button.
```

Original comment by `christian.a.reiter@gmail.com` on 30 Aug 2012 at 12:58

**Comment by GoogleCodeExporter** [comment]
```
yes, but you think as a geek, not as a user.
A user click.
No need to think to clic.
Otherwise, shortcuts need to think.

And, can you say that the rest of the FMF soft is usable ?
And can't you see that accountancy is a complex procedure, and difficult to 
manage to the user, especially a GP in the few seconds he has  to do so.
I am a GP, and i say that rapidocmptamed was more easy to use, with a 
percentage of errors low. Because of few actions to record the values, because 
of big buttons.
But Eric don't like big multipurpose buttons, i had to manage with that ...

```

Original comment by `pm.desom...@gmail.com` on 30 Aug 2012 at 6:16

**Comment by GoogleCodeExporter** [comment]
```
I think as a geek AND as a GP, hehe ;-)
Oh, and this is no geek thinking, this is Human Interface Design Guidelines!

And I've worked at some GPs that were using that old system, and were VERY 
happy because they were fast...
When I e.g. got rapidocomptamed, first I couldn't compile it, then I did not 
see how to use it and needed a bit advice from you: this is the best example 
for it: the software maybe is VERY efficient (because YOU programmed it 
according to YOUR needs and you know where to click and how wo use it fast!), 
but NOT very usable - because I  (in this case as "user", no geek, did not find 
out how to use it, found many glitches, etc.).
This is normal, because programmers tend to program for their owns. Just 
because you (in this case as programmer!) can use your own software, it does 
NOT mean that anyone else can use it too.

Therefore there are
* Standards 
(e.g. where buttons should be - even in Mac/Windows/Gnome/KDE/XFCE are button 
orders (Ok, Cancel, etc) not the same - but inside ONE OS they should be 
consistent - so it's a design fault to always put the ok button on the right 
e.g. - therefore we have the QButtonBox!)
* design guidelines

I think that the current system in Accountancy is not easy to use.
I already saw a system in a proprietary software (the rest of the software was 
afwul!) that was MUCH easier.

So I say that usability and efficiency are TWO different things to consider, 
and BOTH have to be implemented in a good software!
```

Original comment by `christian.a.reiter@gmail.com` on 30 Aug 2012 at 6:31

**Comment by GoogleCodeExporter** [comment]
```
saw your changes - already MUCH better.
I changed the toggle buttons to the left - if that is ok for you - so that the 
visual is at least like it was a toggle button and the user can imagine what 
happens when he clicks on it. Please tell my if that is ok for you, Pierre.
http://code.google.com/p/freemedforms/source/detail?r=f207f7287c4607a6c0c56bf06f
47f26ba567dd70
```

Original comment by `christian.a.reiter@gmail.com` on 30 Aug 2012 at 7:25

**Comment by GoogleCodeExporter** [comment]
```
ok for the button ...
Am i the first to say that you are a stubborn geek ?
;)

```

Original comment by `pm.desom...@gmail.com` on 31 Aug 2012 at 6:59

**Comment by GoogleCodeExporter** [comment]
```
Hmm, eh, ye... no. ;-)

I just saw many software at collegues' practices that was full of stupid errors 
that caused the GPs to curse the program all the time. So many things that were 
*clearly* designed by a programmer and never used by any doctor before 
launching.

And I am a bit of a perfectionist in that case, yes.
But my dream had always been to work with (and contribute to) a software that 
is usable AND efficient, AND Open Source.

So my way is suggesting standards and good workflows everywhere I can see.
Maybe call me stubborn ;-) but many others will thank you when they use the 
billing plugin and can actually understand what it does ;-)
```

Original comment by `christian.a.reiter@gmail.com` on 1 Sep 2012 at 10:49

**Comment by GoogleCodeExporter** [comment]
Original comment by `eric.mae...@gmail.com` on 25 Sep 2012 at 4:20
- Added labels: **Milestone-Release0.8.0**, **Priority-Critical**
- Removed labels: **Priority-Low**

**Comment by GoogleCodeExporter** [comment]
Original comment by `christian.a.reiter@gmail.com` on 14 Nov 2012 at 2:41

**Comment by GoogleCodeExporter** [comment]
Original comment by `christian.a.reiter@gmail.com` on 13 Dec 2012 at 11:57
- Now blocked on: #194

**Comment by GoogleCodeExporter** [comment]
Original comment by `eric.mae...@gmail.com` on 12 Feb 2013 at 9:07
- Added labels: **Milestone-Release1.0**
- Removed labels: **Milestone-Release0.8.0**

**Comment by GoogleCodeExporter** [comment]
Original comment by `eric.mae...@gmail.com` on 15 Sep 2013 at 5:08
- Changed state: **WontFix**
